### ISSUE TITLE
DaemonSet consul nodes come up as single clusters

### ISSUE THREAD

**Issue by mellotron** [cilium/cilium#832]
Hello,

When I deploy consul and cilium across a multi-node cluster with:

```
$ kubectl create -f https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/cilium-ds-rbac.yaml
```

I see the pods come up:

```
kube-system   cilium-bgwc0                    1/1       Running   0          3h        192.168.1.9    node2
kube-system   cilium-consul-ghph3             1/1       Running   0          8m        192.168.1.5    node1
kube-system   cilium-consul-rf3rp             1/1       Running   0          3h        192.168.1.9    node2
kube-system   cilium-hrt9w                    1/1       Running   0          8m        192.168.1.5    node1
```

However it looks like the consuls have formed single node clusters:

```
$ kubectl exec -n kube-system cilium-consul-rf3rp -it consul members
Node    Address         Status  Type    Build  Protocol  DC
node2  127.0.0.1:8301  alive   server  0.6.4  2         dc1
```

```
$ kubectl exec -n kube-system cilium-consul-ghph3 -it consul members
Node   Address         Status  Type    Build  Protocol  DC
node1  127.0.0.1:8301  alive   server  0.6.4  2         dc1
```

```
$ kubectl logs -n kube-system cilium-consul-ghph3
==> WARNING: BootstrapExpect Mode is specified as 1; this is the same as Bootstrap mode.
==> WARNING: Bootstrap mode enabled! Do not enable unless necessary
==> Starting Consul agent...
==> Starting Consul agent RPC...
==> Consul agent running!
         Node name: 'node1'
        Datacenter: 'dc1'
            Server: true (bootstrap: true)
       Client Addr: 127.0.0.1 (HTTP: 8500, HTTPS: -1, DNS: 8600, RPC: 8400)
      Cluster Addr: 127.0.0.1 (LAN: 8301, WAN: 8302)
    Gossip encrypt: false, RPC-TLS: false, TLS-Incoming: false
             Atlas: <disabled>

==> Log data will now stream in as it occurs:

    2017/05/27 14:42:19 [INFO] raft: Node at 127.0.0.1:8300 [Follower] entering Follower state
    2017/05/27 14:42:19 [INFO] serf: EventMemberJoin: node1 127.0.0.1
    2017/05/27 14:42:19 [INFO] consul: adding LAN server node1 (Addr: 127.0.0.1:8300) (DC: dc1)
    2017/05/27 14:42:19 [INFO] serf: EventMemberJoin: node1.dc1 127.0.0.1
    2017/05/27 14:42:19 [INFO] consul: adding WAN server node1.dc1 (Addr: 127.0.0.1:8300) (DC: dc1)
    2017/05/27 14:42:19 [ERR] agent: failed to sync remote state: No cluster leader
    2017/05/27 14:42:20 [WARN] raft: Heartbeat timeout reached, starting election
    2017/05/27 14:42:20 [INFO] raft: Node at 127.0.0.1:8300 [Candidate] entering Candidate state
    2017/05/27 14:42:20 [INFO] raft: Election won. Tally: 1
    2017/05/27 14:42:20 [INFO] raft: Node at 127.0.0.1:8300 [Leader] entering Leader state
    2017/05/27 14:42:20 [INFO] consul: cluster leadership acquired
    2017/05/27 14:42:20 [INFO] consul: New leader elected: node1
    2017/05/27 14:42:20 [INFO] raft: Disabling EnableSingleNode (bootstrap)
    2017/05/27 14:42:20 [INFO] consul: member 'node1' joined, marking health alive
    2017/05/27 14:42:21 [INFO] agent: Synced service 'consul'
==> Newer Consul version available: 0.8.3
```

**Comment by mellotron** [comment]
Potentially can be done as a Job:

- https://github.com/kelseyhightower/consul-on-kubernetes/blob/master/jobs/consul-join.yaml

**Comment by aanm** [comment]
@mellotron The `cilium-ds-rbac.yaml` file that you initially described no longer exists and was replaced by `https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/cilium-ds-gsg.yaml` which is intended to be used on a single node. Otherwise you end up with a single consul cluster node like you've describe.

Since kubernetes uses etcd to store data, we have created a new daemon set file to use etcd as well. Just make sure you're running etcd >= 3.1.0.

The URL of the new Daemon Set is this one: `https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/cilium-ds.yaml`

The instructions to use that file are described here: http://docs.cilium.io/en/latest/admin/#ds-config
